Wikipedia/Code/Session.swift

```diff
@@ -263,4 +265,11 @@ import Foundation
             completion(summaryResponses)
         }
     }
+    
+    private func addAppInstallIDIfNeeded(_ request: inout URLRequest) {
+        guard SessionSingleton.sharedInstance().shouldSendUsageReports, let appInstallID = UserDefaults.wmf_userDefaults().wmf_appInstallID else {
```

Comment:
If this header needs to be set on all requests, it should be done through`URLSession.shared.configuration.httpAdditionalHeaders` - that way it covers the places in the code we're still using `URLSession.shared`  directly. There are also manually created `URLSession`s in `ImageController` and `EventLoggingService` that might need to be updated